src/colvarbias_opes.cpp

```diff
@@ -0,0 +1,1977 @@
+  const colvarbias_opes::kernel& G,
+  const std::vector<cvm::real>& x,
+  std::vector<cvm::real>& accumulated_derivative,
+  std::vector<cvm::real>& dist) const {
+  cvm::real norm2 = 0;
+  for (size_t i = 0; i < num_variables(); ++i) {
+    dist[i] = 0.5 * variables(i)->dist2_lgrad(x[i], G.m_center[i]) / G.m_sigma[i];
+    norm2 += dist[i] * dist[i];
+    if (norm2 >= m_cutoff2) {
+      return 0;
+    }
+  }
+  const cvm::real val = G.m_height * (std::exp(-0.5 * norm2) - m_val_at_cutoff);
+  // The derivative of norm2 with respect to x
+  for (size_t i = 0; i < num_variables(); ++i) {
+    accumulated_derivative[i] -= val * dist[i] / G.m_sigma[i];
+  }
+  return val;
+}
+
+cvm::real colvarbias_opes::getProbAndDerivatives(
+  const std::vector<cvm::real>& cv, std::vector<cvm::real>& der_prob) const {
+  double prob = 0.0;
+  std::vector<cvm::real> dist(num_variables(), 0);
+  if (!m_nlist) {
+    if (m_num_threads == 1 || m_kernels.size() < 2 * m_num_threads) {
+      for (size_t k = 0; k < m_kernels.size(); ++k) {
+        prob += evaluateKernel(m_kernels[k], cv, der_prob, dist);
+      }
+    } else {
+#if defined(_OPENMP)
+      #pragma omp parallel num_threads(m_num_threads)
+      {
+        std::vector<cvm::real> omp_deriv(der_prob.size(), 0);
+        std::vector<cvm::real> tmp_dist(num_variables());
+        #pragma omp for reduction(+:prob) nowait
+        for (size_t k = 0; k < m_kernels.size(); ++k) {
+          prob += evaluateKernel(m_kernels[k], cv, omp_deriv, tmp_dist);
+        }
+        #pragma omp critical
+        for (size_t i = 0; i < num_variables(); ++i) {
+          der_prob[i]+=omp_deriv[i];
+        }
+        #pragma omp single
+        for (size_t i = 0; i < num_variables(); ++i) {
+          dist[i] = tmp_dist[i];
+        }
+      }
+#elif defined(CMK_SMP) && defined(USE_CKLOOP)
+      // TODO: Does this work??
+      std::vector<std::vector<cvm::real>> derivs(m_num_threads, std::vector<cvm::real>(num_variables(), 0));
+      std::vector<std::vector<cvm::real>> dists(m_num_threads, std::vector<cvm::real>(num_variables(), 0));
+      auto worker = [&](int start, int end, void* result){
+        const int tid = cvm::proxy->smp_thread_id();
+        double tmp_prob = 0;
+        for (int i = start; i <= end; ++i) {
+          tmp_prob += evaluateKernel(m_kernels[i], cv, derivs[tid], dists[tid]);
+        }
+        *(double *)result = tmp_prob;
+      };
+      const size_t numChunks = m_kernels.size();
+      const size_t lowerRange = 0;
+      const size_t upperRange = numChunks - 1;
+      CkLoop_Parallelize(
+        numChunks, lowerRange, upperRange,
+        worker, &prob, CKLOOP_DOUBLE_SUM, NULL);
+      for (size_t i = 0; i < num_variables(); ++i) {
+        for (size_t j = 0; j < m_num_threads; ++j) {
+          if (j == 0) dist[i] = dists[j][i];
+          der_prob[i] += derivs[j][i];
+        }
+      }
+#else
+      cvm::error("Unsupported threading library.\n");
```

Comment:
How about: "OPES cannot run because this binary is not linked with a supported threading library".